packages/hardhat-chai-matchers/src/emit.ts

```diff
@@ -87,26 +89,35 @@ export function supportEmit(
 export async function emitWithArgs(
   context: any,
   Assertion: Chai.AssertionStatic,
+  chaiUtils: Chai.ChaiUtils,
   expectedArgs: any[]
 ) {
-  tryAssertArgsArraysEqual(context, Assertion, expectedArgs, context.logs);
+  tryAssertArgsArraysEqual(
+    context,
+    Assertion,
+    chaiUtils,
+    expectedArgs,
+    context.logs
+  );
 }
 
 function assertArgsArraysEqual(
   context: any,
   Assertion: Chai.AssertionStatic,
+  chaiUtils: Chai.ChaiUtils,
   expectedArgs: any[],
   log: any
 ) {
   const { utils } = require("ethers");
 
-  const actualArgs = (context.contract.interface as Interface).parseLog(
-    log
-  ).args;
+  const actualArgs = (
+    chaiUtils.flag(context, "contract").interface as Interface
```

Comment:
I think using the flags inline here (and in some other places in the PR) makes the code a little harder to read, can we extract them to some variables?

Reply:
https://github.com/NomicFoundation/hardhat/pull/2741/commits/290ac34ed88bdd1bca38420b82445c5caa2b9c3b